chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -107,20 +107,26 @@ impl EthereumAdapter {
     ) -> Self {
         let web3 = Arc::new(Web3::new(transport));
 
-        info!(logger, "Checking if provider supports getBlockReceipts"; "provider" => &provider);
+        info!(logger, "Checking if provider supports getBlockReceipts");
 
         // Check if the provider supports `getBlockReceipts` method.
-        let supports_block_receipts = Self::check_block_receipt_support(
-            web3.clone(),
-            &provider,
-            supports_eip_1898,
-            call_only,
-            &logger,
-        )
-        .await
-        .is_ok();
-
-        info!(logger, "Checked if provider supports eth_getBlockReceipts"; "provider" => &provider, "supports_block_receipts" => supports_block_receipts);
+        let block_receipts_support_result =
+            Self::check_block_receipt_support(web3.clone(), supports_eip_1898, call_only, &logger)
```

Comment:
the fact this happens at the start means we probably need to set an upper limit on how long we wait, if this is a 1m timeout on start it will add minutes to the total start-up time if a provider is down. Maybe we need to check this lazily?

Reply:
noted, making that change, Also should we add a few retries as well?

Reply:
Makes sense, a lazy check or a stateful check is the long term solution for sure. Let me look into making this a lazy loading one.